cmd/servicelog/list.go

```diff
@@ -1,123 +1,123 @@
-		return err
+	if err = printServiceLogResponse(response); err != nil {
+		return fmt.Errorf("failed to print service logs: %w", err)
 	}
+
 	return nil
 }
 
 var serviceLogListAllMessagesFlag = false
 var serviceLogListInternalOnlyFlag = false
 
-func FetchServiceLogs(clusterID string) (*sdk.Response, error) {
-	// Create OCM client to talk to cluster API
-	ocmClient, err := utils.CreateConnection()
-	if err != nil {
-		return nil, err
+func printServiceLogResponse(response *slv1.ClustersClusterLogsListResponse) error {
+	entryViews := logEntryToView(response.Items().Slice())
+	view := LogEntryResponseView{
+		Items: entryViews,
+		Kind:  "ClusterLogList",
+		Page:  response.Page(),
+		Size:  response.Size(),
+		Total: response.Total(),
 	}
-	defer func() {
-		if err := ocmClient.Close(); err != nil {
-			fmt.Printf("Cannot close the ocmClient (possible memory leak): %q", err)
-		}
-	}()
 
-	// Use the OCM client to retrieve clusters
-	clusters := utils.GetClusters(ocmClient, []string{clusterID})
-	if len(clusters) != 1 {
-		return nil, fmt.Errorf("GetClusters expected to return 1 cluster, got: %d", len(clusters))
+	viewBytes, err := json.Marshal(view)
+	if err != nil {
+		return fmt.Errorf("failed to marshal response for output: %w", err)
 	}
-	cluster := clusters[0]
 
-	// Now get the SLs for the cluster
-	return utils.SendRequest(CreateListSLRequest(ocmClient, cluster, serviceLogListAllMessagesFlag, serviceLogListInternalOnlyFlag))
+	return dump.Pretty(os.Stdout, viewBytes)
 }
 
-func init() {
-	// define required flags
-	listCmd.Flags().BoolVarP(&serviceLogListAllMessagesFlag, "all-messages", "A", serviceLogListAllMessagesFlag, "Toggle if we should see all of the messages or only SRE-P specific ones")
-	listCmd.Flags().BoolVarP(&serviceLogListInternalOnlyFlag, "internal", "i", serviceLogListInternalOnlyFlag, "Toggle if we should see internal messages")
+type LogEntryResponseView struct {
+	Items []*LogEntryView `json:"items"`
+	Kind  string          `json:"kind"`
+	Page  int             `json:"page"`
+	Size  int             `json:"size"`
+	Total int             `json:"total"`
 }
 
-func CreateListSLRequest(ocmClient *sdk.Connection, cluster *cmv1.Cluster, allMessages bool, internalMessages bool) *sdk.Request {
-	// Create and populate the request:
-	request := ocmClient.Get()
-	err := arguments.ApplyPathArg(request, targetAPIPath)
-	if err != nil {
-		log.Fatalf("Can't parse API path '%s': %v\n", targetAPIPath, err)
-	}
-	var empty []string
-
-	// prefer cluster external over cluster internal ID
-	var formatMessage string
-	if cluster.ExternalID() != "" {
-		formatMessage = fmt.Sprintf(`search=cluster_uuid = '%s'`, cluster.ExternalID())
-	} else {
-		formatMessage = fmt.Sprintf(`search=cluster_id = '%s'`, cluster.ID())
-	}
+// TODO uncomment these fields after https://github.com/openshift-online/ocm-sdk-go/pull/847 is merged
+type LogEntryView struct {
```

Comment:
**Question:** Why can't we use the `slv1.LogEntry` type directly? Is it to have everything as string for a proper print format after `viewBytes, err := json.Marshal(view)`?

Reply:
We can't marshal it because the fields aren't exported unfortunately. This is the struct:
```go
type LogEntry struct {
	bitmap_        uint32
	id             string
	href           string
	clusterID      string
	clusterUUID    string
	description    string
	eventStreamID  string
	logType        LogType
	serviceName    string
	severity       Severity
	subscriptionID string
	summary        string
	timestamp      time.Time
	username       string
	internalOnly   bool
}
```

After [chatting with the OCM team,](https://redhat-internal.slack.com/archives/CBDNMS43V/p1695676203847559), this was determined to be the best path forward.